### PR TITLE
Update Babel to Current Standard "babel-preset-env"

### PR DESCRIPTION
**Pull request by ajnauleau**
Install babel dependencies by running

```
npm install babel-cli -D
npm install babel-preset-env -D
```

updated .babelrc file inside your project and add the following code inside it:

```json
{
  "presets": ["env"]
}
````

Update package.json and add same code within, while keeping current settings.

Change the order of presets to "env", "react", "stage-0".

Build, import into extensions, and tested.

Fixed.

### CODE REVIEW

**Comment by bdresser** [comment]
@whymarrh or @alextsg could you take a look when you get a sec

**Comment by whymarrh** [comment]
@ajnauleau thanks! I'll take a look at this in a bit.

File: package.json [hunk 1]
```diff
@@ -54,10 +54,17 @@
         "babelify",
         {
           "presets": [
-            "es2015",
-            "stage-0"
+            ["env", {
```
Comment (whymarrh) [inline package.json]: I think we might be better off removing the options here and having it default to using what's in the `.babelrc`

File: package.json [hunk 2]
```diff
@@ -203,18 +210,20 @@
...
     "babel-polyfill": "^6.23.0",
+    "babel-preset-env": "^1.7.0",
     "babel-preset-react": "^6.24.1",
     "babel-preset-stage-0": "^6.24.1",
     "babel-register": "^6.7.2",
     "babelify": "^8.0.0",
     "beefy": "^2.1.5",
     "brfs": "^1.6.1",
-    "browserify": "^16.1.1",
+    "browserify": "^16.2.2",
```
Comment (whymarrh) [inline package.json]: Can we drop the changes to non-babel packages? (We could move them to a different PR if we want to bump them independently?)

Comment (whymarrh) [reply package.json]: See also: `ganache-core` and `node-sass` below

**Review by whymarrh — CHANGES_REQUESTED** [review]
Two small things and can we make sure that we regenerate the `package-lock.json` from npm@6 on Node 8

File: package.json (hunk 1 above)
Comment (ajnauleau) [reply package.json]: Seems like Babelify doesn't recognise .babelrc anymore with babel@6, so there wouldn't actually be any need to update .babelrc, but the package.json file is still necessary. I think we still should keep .babelrc updated, incase it could have other uses in the future.

https://github.com/babel/babelify/issues/151

https://github.com/babel/babelify/issues/151

**Comment by ajnauleau** [comment]
Gulp build, and testing on chrome worked on my local environment, but once pushed to PR didn't pass **circleCI**. Removed commit regenerating `package-lock.json` and pushed to PR, checks out now. Any suggestions on regenerating `package-lock.json` file?

**Comment by ajnauleau** [comment]
Bugs seems to have been fixed, worth reviewing now @whymarrh

**Comment by ajnauleau** [comment]
Circle-CI isn't running `npm install` before building, `missing node_modules` -- build works locally, will keep looking into this.

**Comment by brunobar79** [comment]
@ajnaleau make sure you are using node 8.11.3 and npm 6.1.0. If you are using nvm, you can type nvm use and it will switch automatically to the right version.

Once you do that, nuke your node_modules folder, run npm install and commit / push the updated package-lock.json

That should fix the issue. I’ll update the docs later today.

**Comment by ajnauleau** [comment]
@brunobar79 updated node and npm with nvm but I don't think that was the issue. The tests aren't running `npm install` before running , therefore the modules such as `gulp` are missing. 

Scratching my head on this one...

**Comment by brunobar79** [comment]
@ajnauleau  npm install IS running ([see here](https://circleci.com/gh/MetaMask/metamask-extension/39899?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)) but there's a diff generated on the package-lock.json after running npm install on the CI server that makes the build to restore the wrong cache.

That's usually because the package-lock.json is not updated or the node/npm version is incorrect.
Anyway, this issue became very annoying for a lot of people so I just opened https://github.com/MetaMask/metamask-extension/pull/4847 which should take care of the issue.

I'll ping you here once that makes it to develop and you can pull develop, re-run npm i commit and push and I'm sure the build will pass

**Comment by ajnauleau** [comment]
@brunobar79 after reviewing the tests, it seems like circle-ci was reverting to a cached version on the server. Your fix in #4847 seems to resolve it-- will wait until your update, thanks.

**Comment by brunobar79** [comment]
@ajnauleau my PR just got merged.

File: package.json (hunk 2 above)
Comment (ajnauleau) [reply package.json]: @whymarrh - Solved as well, looks like the PR is good to merge.

**Comment by ajnauleau** [comment]
@whymarrh - I'm unable to review the requested changes you made, would you be able to check this out? Seems all good to go.

Thanks!

**Comment by whymarrh** [comment]
Awesome, thanks @ajnauleau! This LGTM and I don't think it increases our (already slow) build time.

I can confirm that the dist builds (`npm run dist`) are still being browserified (refs #3628 and #3657).

**Review by whymarrh — APPROVED** [review]
